airbyte-webapp/src/packages/cloud/components/experiments/FreeConnectorProgram/hooks/useFreeConnectorProgram.ts

```diff
@@ -1,26 +1,26 @@
 import { useQuery } from "react-query";
 
+import { useExperiment } from "hooks/services/Experiment";
 import { useConfig } from "packages/cloud/services/config";
 import { useDefaultRequestMiddlewares } from "services/useDefaultRequestMiddlewares";
 import { useCurrentWorkspaceId } from "services/workspaces/WorkspacesService";
 
 import { webBackendGetFreeConnectorProgramInfoForWorkspace } from "../lib/api";
 
-export const useFreeConnectorProgramInfo = () => {
+export const useFreeConnectorProgram = () => {
   const workspaceId = useCurrentWorkspaceId();
   const { cloudApiUrl } = useConfig();
   const config = { apiUrl: cloudApiUrl };
   const middlewares = useDefaultRequestMiddlewares();
   const requestOptions = { config, middlewares };
+  const freeConnectorProgramEnabled = useExperiment("workspace.freeConnectorsProgram.visible", false);
 
   return useQuery(["freeConnectorProgramInfo", workspaceId], () =>
     webBackendGetFreeConnectorProgramInfoForWorkspace({ workspaceId }, requestOptions).then(
       ({ hasEligibleConnector, hasPaymentAccountSaved }) => {
-        const showEnrollmentUi = !hasPaymentAccountSaved && hasEligibleConnector;
-        // TODO hardcoding this value to allow testing while data source gets sorted out
-        const needsEmailVerification = false;
+        const eligibleToEnroll = !hasPaymentAccountSaved && hasEligibleConnector;
```

Comment:
I mean, wouldn't a user technically be ineligible if they don't have the experiment on?

Reply:
You're not wrong! The intended meaning was more like "meets eligibility requirements", but really I just wanted to define the business logic part of the combined condition as its own thing. LMK if you think it would be clearer with a different name.

Reply:
> I think it'd be cleaner on the other end to just have something that tells the UI "should we show this thing"?

Perhaps I'm misunderstanding, but that's is what the function returns: a single boolean that tells the UI whether to show a thing. It would behave identically if we combined all the input conditions in one big boolean expression, `freeConnectorProgramEnabled && !hasPaymentAccountSaved && hasEligibleConnector`; the fact that the function gives a separate name to the `!hasPaymentAccountSaved && hasEligibleConnector` subset of that is an internal implementation detail.

Reply:
disregard.  i've read this file too many times.